### PR TITLE
[jsruntime][BreakChange] add runtime code (a/b) and build success

### PR DESCRIPTION
**Pull request by lucky-chen**
[Android] rebuild jsc for initHeapTimer and jsruntme/old  code are work fine

[jsruntime] dom_wson switch fix

[Android] add args check fix && add a/b so load/update logic

[Android] build logic for a/b and __enable_native_promise__ flag

Fix License issue for javaScriptCore (#2498)

License under following directory:
* `weex_core/Source/include/JavaScriptCore`
* `weex_core/Source/include/wtf`

Update license

Update Android Build Script and License.

[Android] update jsc.so

[Android] update jsfm to 0.27

[Android] add data record

[Android] add build script for build runtime-jss.so

load so dynamic

[Android] fix icu init

fix v7a build

[Android] fix jsfm switch promise flag

### CODE REVIEW

File: android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java [hunk 1]
```diff
@@ -304,6 +307,61 @@ public static void copyStartUpSo() {
     }
   }
 
+  public static void copyJssRuntimeSo(){
+    Log.e("test->", "enter copyJssRuntimeSo: ");
```
Comment (YorkShen) [inline android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java]: Please remove `Log.e` or change them to WXLogUtils.e

File: dangerfile.js [hunk 2]
```diff
@@ -246,7 +246,9 @@ const ignoreCopyrightVerifyPath = [
   'ios/playground/bundlejs',
   'ios/sdk/WeexSDK/Resources',
   'ios/sdk/WeexSDK/Sources/Layout',
-  'ios/sdk/WeexSDK/dependency/SRWebSocket'
+  'ios/sdk/WeexSDK/dependency/SRWebSocket',
+  'weex_core/Source/include',
+  'weex_core/Source/base/third_party'
```
Comment (YorkShen) [inline dangerfile.js]: Change to `'weex_core/Source/base/third_party/icu'`

File: dangerfile.js [hunk 3]
```diff
@@ -246,7 +246,9 @@ const ignoreCopyrightVerifyPath = [
   'ios/playground/bundlejs',
   'ios/sdk/WeexSDK/Resources',
   'ios/sdk/WeexSDK/Sources/Layout',
-  'ios/sdk/WeexSDK/dependency/SRWebSocket'
+  'ios/sdk/WeexSDK/dependency/SRWebSocket',
+  'weex_core/Source/include',
```
Comment (YorkShen) [inline dangerfile.js]: Change to

```
'weex_core/Source/include/JavaScriptCore',
'weex_core/Source/include/wtf'
```

Comment (lucky-chen) [reply dangerfile.js]: done

File: dangerfile.js (hunk 2 above)
Comment (lucky-chen) [reply dangerfile.js]: done

File: android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java (hunk 1 above)
Comment (lucky-chen) [reply android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java]: done

File: android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java [hunk 4]
```diff
@@ -304,6 +307,60 @@ public static void copyStartUpSo() {
     }
   }
 
+  public static void copyJssRuntimeSo(){
+    boolean tryUseRunTimeApi = WXUtils.checkGreyConfig("wxapm","use_runtime_api","100");
+    Log.e("weex", "tryUseRunTimeApi ? "+ tryUseRunTimeApi);
```
Comment (YorkShen) [inline android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java]: Delete it or change to `WXLogUtils.i`.

File: android/sdk/src/main/java/com/taobao/weex/utils/WXUtils.java [hunk 5]
```diff
@@ -569,6 +572,25 @@ public static int getNumberInt(Object value, int defaultValue){
     }catch (Exception e){return  defaultValue;}
   }
 
+  public static boolean checkGreyConfig(String group,String key,String defaultValue){
+      Log.e("test->", "enter checkGreyConfig");
```
Comment (YorkShen) [inline android/sdk/src/main/java/com/taobao/weex/utils/WXUtils.java]: Delete it or change to `WXLogUtils.i`
* Line 576
* Line 579
* Line 590

Comment (lucky-chen) [reply android/sdk/src/main/java/com/taobao/weex/utils/WXUtils.java]: done

File: android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java (hunk 4 above)
Comment (lucky-chen) [reply android/sdk/src/main/java/com/taobao/weex/utils/WXSoInstallMgrSdk.java]: done

**Comment by c0decafe** [comment]
note that this pull breaks the ios sdk build for me https://github.com/apache/incubator-weex/issues/2673

**Comment by YorkShen** [comment]
Already fixed  @c0decafe

**Pull request merged**